core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -125,6 +126,49 @@ public String toString() {
     }
   }
 
+  public static class MetadataLogEntry {
+    private final long timestampMillis;
+    private final String file;
+
+    MetadataLogEntry(long timestampMillis, String file) {
+      this.timestampMillis = timestampMillis;
+      this.file = file;
+    }
+
+    public long timestampMillis() {
+      return timestampMillis;
+    }
+
+    public String file() {
+      return file;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+      if (this == other) {
+        return true;
+      } else if (!(other instanceof MetadataLogEntry)) {
+        return false;
+      }
+      MetadataLogEntry that = (MetadataLogEntry) other;
+      return timestampMillis == that.timestampMillis &&
+              java.util.Objects.equals(file, that.file);
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hashCode(timestampMillis, file);
+    }
+
+    @Override
+    public String toString() {
+      return MoreObjects.toStringHelper(this)
+              .add("timestampMillis", timestampMillis)
```

Comment:
nit: seems like this file is still formatted with 8 spaces for continued indentation.